src/logging.h

```diff
@@ -104,7 +109,7 @@ namespace BCLog {
         std::atomic<Level> m_log_level{DEFAULT_LOG_LEVEL};
 
         /** Log categories bitfield. */
-        std::atomic<uint32_t> m_categories{0};
+        LogFlagsBitset m_categories GUARDED_BY(m_cs);
```

Comment:
Tested approach ACK. I haven't yet checked impact on performance with the benchmarks.

Reply:
Switching this from an atomic to be blocking on the `m_cs` mutex doesn't seem great? It's checked for every `LogPrintf` where the log category is disabled, etc. Should at least use a reader/writer lock?